File: common/src/main/java/work/fking/common/Rand.java
package work.fking.common;

import lombok.experimental.UtilityClass;

import java.security.SecureRandom;
import java.util.Random;

@UtilityClass
public class Rand {

    private static final Random RANDOM = new SecureRandom();

    public static String randomHexString(int length) {
        byte[] bytes = new byte[length];
        RANDOM.nextBytes(bytes);

        StringBuilder builder = new StringBuilder();

        for (byte aByte : bytes) {
            builder.append(String.format("%02x", aByte));
        }
        return builder.toString();
    }

    public static long nextLong() {
        return RANDOM.nextLong();
    }

    /**
     * Returns a value between zero and the given max value.
     *
     * @param maximum The maximum value, inclusive.
     * @return A value between zero and the max value.
     */
    public static int withMax(int maximum) {
        return between(0, maximum);
    }

    /**
     * Returns a pseudo-random value between the given bounds.
     *
     * @param minimum The minimum value, inclusive.
     * @param maximum The maximum value, inclusive.
     * @return The random value.
     */
    public static int between(int minimum, int maximum) {

        if (minimum > maximum) {
            throw new IllegalArgumentException("The minimum value must be less than the maximum");
        }
        return minimum + RANDOM.nextInt(maximum - minimum + 1);
    }
}

File: game-server/src/main/kotlin/work/fking/pangya/game/Rand.kt
package work.fking.pangya.game

import java.security.SecureRandom
import java.util.Random

object Rand {
    private val RANDOM: Random = SecureRandom()

    fun randomHexString(length: Int): String {
        val bytes = ByteArray(length)
        RANDOM.nextBytes(bytes)
        val builder = StringBuilder()
        for (aByte in bytes) {
            builder.append(String.format("%02x", aByte))
        }
        return builder.toString()
    }

    fun nextInt(): Int {
        return RANDOM.nextInt()
    }

    fun nextLong(): Long {
        return RANDOM.nextLong()
    }

    fun max(maximum: Int): Int {
        return between(0, maximum - 1)
    }

    fun maxInclusive(maximum: Int): Int {
        return between(0, maximum)
    }

    fun between(minimum: Int, maximum: Int): Int {
        require(minimum <= maximum) { "The minimum value must be less than the maximum" }
        return minimum + RANDOM.nextInt(maximum - minimum + 1)
    }
}
